RWS blog (#7213)
GoogleChrome/developer.chrome.com: site/en/blog/related-website-sets/index.md

---
layout: 'layouts/blog-post.njk'
title: Related Website Sets - the new name for First-Party Sets in Chrome 117
description: >

>   Related Website Sets (RWS) is the new name for First-Party Sets and also brings increased flexibility in defining sets.

Reviewer: It might be worth splitting this to two sentences, because it reads a bit odd now.

>   Related Website Sets (RWS) is the new name for First-Party Sets and also brings increased flexibility in defining sets.

Reviewer: ```suggestion
  Related Website Sets (RWS) is the new name for First-Party Sets. It also brings increased flexibility in defining sets.
```

* The [ccTLD use case](https://github.com/WICG/first-party-sets/tree/main#defining-a-set-through-use-case-based-subsets) addresses breakages like the [login example](https://issuetracker.google.com/268390722) filed in our public tracker. Such cases are often addressed in the ecosystem through heuristic-based exceptions (See [ref 1](#ref-1)).
* The [service domain use case](https://github.com/WICG/first-party-sets/tree/main#defining-a-set-through-use-case-based-subsets) addresses a common developer practice to isolate sensitive functions (like supporting an authentication flow) from user-facing domains. Such cases may be addressed in the ecosystem through targeted [exceptions](https://searchfox.org/mozilla-central/rev/3002762e41363de8ee9ca80196d55e79651bcb6b/browser/extensions/webcompat/data/shims.js#686) (See [ref 2](#ref-2)).

> *   The [associated domain use case](https://github.com/WICG/first-party-sets/tree/main#defining-a-set-through-use-case-based-subsets) provides more flexibility for the types of domains that may require third-party cookie access for critical user journeys (See [ref 3](#ref-3)). While the ccTLD and service domain use cases employ strict technical checks based on domain characteristics to minimize abuse, the associated domain uses a numeric limit. Read more about this below.

Reviewer: ```suggestion
*   The [associated domain use case](https://github.com/WICG/first-party-sets/tree/main#defining-a-set-through-use-case-based-subsets) provides more flexibility for the types of domains that may require third-party cookie access for critical user journeys (See [ref 3](#ref-3)). While the ccTLD and service domain use cases employ strict technical checks based on domain characteristics to minimize abuse, the associated domain uses a numeric limit. Read more about this in the following sections.
```

## Associated domain limit increased to five domains 

> Chrome previously proposed a numeric limit of 3 domains for the Associated Subset (plus one primary domain), in alignment with our objective to prevent widespread tracking abuse. We have heard feedback from web standards participants that the limit was too low for different types of use cases. 

Reviewer: ```suggestion
Chrome previously proposed a numeric limit of three domains for the Associated Subset (plus one primary domain), in alignment with our objective to prevent widespread tracking abuse. We have heard feedback from web standards participants that the limit was too low for different types of use cases. 
```

> We have decided to increase the associated domain limit to 5 domains (plus one primary domain) which best matches the most comparable implementation offered by another major browser (See [ref 4](#ref-4)). This will take effect beginning in Chrome 117.

Reviewer: ```suggestion
We have decided to increase the associated domain limit to five domains (plus one primary domain) which best matches the most comparable implementation offered by another major browser (See [ref 4](#ref-4)). This will take effect beginning in Chrome 117.
```

Since RWS is not intended as an ads solution, we are not taking into consideration feedback on how to improve RWS to better serve ads use cases. For ads use cases, developers should explore using the Topics, Protected Audience, and Attribution Reporting APIs and provide feedback on them accordingly. 

> ## Users have a choice for extended use cases, beyond 5 associated domains

Reviewer: ```suggestion
## Users have a choice for extended use cases, beyond five associated domains
```

> For user-impacting experiences that are not supported by this limit, Chrome is [working on a user prompting flow](https://github.com/cfredric/chrome-storage-access-api) that also leverages the Storage Access API (SAA), a standard adopted by other browsers. For use cases that need more than 5 associated domains, we encourage developers to evaluate how SAA may be supported in non-RWS contexts. We are following the Blink launch process separately for [this feature](https://groups.google.com/a/chromium.org/g/blink-dev/c/JHf7CWXDZUc/m/Dy2EElgvAgAJ), which will be rolling out in Chrome Desktop beginning in Chrome M117.

Reviewer: ```suggestion
For user-impacting experiences that are not supported by this limit, Chrome is [working on a user prompting flow](https://github.com/cfredric/chrome-storage-access-api) that also leverages the Storage Access API (SAA), a standard adopted by other browsers. For use cases that need more than five associated domains, we encourage developers to evaluate how SAA may be supported in non-RWS contexts. We are following the Blink launch process separately for [this feature](https://groups.google.com/a/chromium.org/g/blink-dev/c/JHf7CWXDZUc/m/Dy2EElgvAgAJ), which will be rolling out in Chrome Desktop beginning in Chrome M117.
```

## Next steps

We're grateful for the ecosystem feedback that has helped shape the API so far. We have invested in RWS as a method of providing developers predictability, control, and agency in preserving the end user experience of the websites they build. We are excited to see how developers adopt and use RWS as we ramp up. The [submission process](https://github.com/GoogleChrome/first-party-sets/blob/main/FPS-Submission_Guidelines.md) is currently live, and the [RWS JSON generator tool](https://goo.gle/rws-json-generator) is a great starting point to help with submissions.

Follow the [Intent-to-Ship thread](https://groups.google.com/a/chromium.org/g/blink-dev/c/7_6JDIfE1as/m/wModmpcaAgAJ) to track progress, and check out [these materials](/en/docs/privacy-sandbox/first-party-sets-integration/) for implementation guidance. 

{% Details %}
{% DetailsSummary %}
References
{% endDetailsSummary %}

> 1. <a id="ref-1"></a>There is general agreement across browsers that these cross-site cookie use-cases are necessary, but have taken different approaches in enabling them. [Firefox](https://developer.mozilla.org/docs/Web/Privacy/Storage_Access_Policy#automatic_storage_access_upon_interaction) ([code](https://searchfox.org/mozilla-central/rev/3002762e41363de8ee9ca80196d55e79651bcb6b/dom/base/Document.cpp#16328)) and [Safari](https://webkit.org/blog/8311/intelligent-tracking-prevention-2-0/#:~:text=Temporary%20Compatibility%20Fix%3A%20Automatic%20Storage%20Access%20for%20Popups) ([code](https://searchfox.org/wubkat/rev/5b368793a8c0a3d99c6991fcd3ef96e3dbd2cf2a/Source/WebKit/NetworkProcess/Classifier/ResourceLoadStatisticsDatabaseStore.cpp#1266)) have both implemented the pop-up heuristic that addresses the breakage observed, for example in the [Nintendo login flow](https://issuetracker.google.com/268390722).

Reviewer: ```suggestion
1. <a id="ref-1"></a>There is general agreement across browsers that these cross-site cookie use-cases are necessary, but that there are different approaches in enabling them. [Firefox](https://developer.mozilla.org/docs/Web/Privacy/Storage_Access_Policy#automatic_storage_access_upon_interaction) ([code](https://searchfox.org/mozilla-central/rev/3002762e41363de8ee9ca80196d55e79651bcb6b/dom/base/Document.cpp#16328)) and [Safari](https://webkit.org/blog/8311/intelligent-tracking-prevention-2-0/#:~:text=Temporary%20Compatibility%20Fix%3A%20Automatic%20Storage%20Access%20for%20Popups) ([code](https://searchfox.org/wubkat/rev/5b368793a8c0a3d99c6991fcd3ef96e3dbd2cf2a/Source/WebKit/NetworkProcess/Classifier/ResourceLoadStatisticsDatabaseStore.cpp#1266)) have both implemented the pop-up heuristic that addresses the breakage observed, for example in the [Nintendo login flow](https://issuetracker.google.com/268390722).
```

Author: Edited the suggestion. Does this scan correctly?

Reviewer: Yep it does.
